lib/gamnit/bmfont.nit

```diff
@@ -406,8 +415,58 @@ class BMFontAsset
 					else 16.0
 				dx += space_advance * scale
 				word_break = true
+			else if c == '[' then
+				# Open link?
+				if i + 1 < text.length and text[i+1] == '[' then
+					# Escape if duplicated
+					i += 1
+				else
+					in_link = true
+					continue
+				end
+			else if c == ']' then
+				# Close link?
+				if i + 1 < text.length and text[i+1] == ']' then
+					# Escape if duplicated
+					i += 1
+				else
+					# If there's a () use it as link_name
+					var j = i + 1
+					if j < text.length and text[j] == '(' then
+						var new_name: nullable String
```

Comment:
yeah, `var a = "foo"` is treated differently from `var b ; b = "foo"`.
The fallback static type of `a` is `String` (is its the most general static type for `a`, so `null` is refused) whereas it is `nullable Object` for b.

Because we don't have union types yet, the fallback static type is used to counter possible loss of static information.